laterpay/asset_sources/js/laterpay-backend-account.js

```diff
@@ -62,32 +75,56 @@
             },
 
             autofocusEmptyInput = function() {
-                var $inputs = $('.lp_js_validateApiKey, .lp_js_validateMerchantId');
-                for (var i = 0, l = $inputs.length; i < l; i++) {
-                    if ($inputs.eq(i).val() === '') {
-                        $inputs.eq(i).focus();
+                var i = 0,
+                    l = $o.apiCredentialsInputs.length;
+
+                for (; i < l; i++) {
+                    if ($o.apiCredentialsInputs.eq(i).val() === '') {
+                        $o.apiCredentialsInputs.eq(i).focus();
                         return;
                     }
                 }
             },
 
+            toggleVisibilityInTestMode = function() {
+                if (hasNoValidCredentials()) {
+                    // FIXME: wtf is this?
+                    $o.testModeInvalidCredField.val(1);
+
+                    // restore test mode
+                    $o.pluginVisibilityToggle.prop('checked', false);
+
+                    // focus Merchant ID input in case the user just forgot to enter his credentials
+                    $o.testMerchantId.focus();
+
+                    // make sure Ajax request gets sent
+                    $o.requestSent = false;
+                } else {
+                    $o.testModeInvalidCredField.val(0);
+                }
+
+                makeAjaxRequest('lp_js_toggleVisibilityInTestModeForm');
+            },
+
             togglePluginModeIndicators = function(mode) {
                 if (mode === 'live') {
-                    $('#lp_js_pluginMode_testText').hide();
-                    $('#lp_js_pluginMode_liveText').show();
-                    $('#lp_js_pluginModeIndicator').fadeOut();
+                    $o.showMerchantContractsButton.fadeOut(250);
+                    $o.pluginModeTestText.hide();
+                    $o.pluginModeLiveText.show();
+                    $o.pluginModeIndicator.fadeOut();
                     $('.lp_liveCredentials').addClass($o.isLive);
                 } else {
-                    $('#lp_js_pluginMode_liveText').hide();
-                    $('#lp_js_pluginMode_testText').show();
-                    $('#lp_js_pluginModeIndicator').fadeIn();
+                    $o.showMerchantContractsButton.fadeIn(250);
+                    $o.pluginModeTestText.show();
+                    $o.pluginModeLiveText.hide();
+                    $o.pluginModeIndicator.fadeIn();
                     $('.lp_liveCredentials').removeClass($o.isLive);
                 }
             },
 
             togglePluginMode = function() {
                 var $toggle                 = $o.pluginModeToggle,
-                    $input                  = $('#lp_js_pluginMode_hiddenInput'),
+                    $input                  = $('#lp_js_pluginMode_hiddenInput'), // FIXME: wtf!? a hidden input for storing the value of a checkbox???
```

Comment:
Screw it, I will merge it now nonetheless, because this is just a formal improvement.
I will create a separate ticket for that.